private/buf/buflsp/file.go

```diff
@@ -1131,6 +1135,25 @@ func (f *file) appendLintErrors(source string, err error) bool {
 	return true
 }
 
+// lineColumnToByteOffset converts a 1-indexed line and 1-indexed byte-based column to a byte offset in the file.
+func (f *file) lineColumnToByteOffset(line, col int) int {
+	currentLine := 1
+	byteOffset := 0
+
+	// Find the start of the target line.
+	for i := 0; i < len(f.text) && currentLine < line; i++ {
```

Comment:
This could be accidentally quadratic :(

Do you have a `report.File` lying around that you can use to avoid this?

Reply:
Fixed, use `InverseLocation` now.